uro98/assignment1/KToLast.java

```diff
@@ -0,0 +1,58 @@
+public class KToLast {
+
+	public static Object kToLastObject(SinglyLinkedList list, int k) {
+
+		SinglyLinkedList.Node node = list.first;
+		int length = 0;
+
+		// Find list length
+		while (node != null) {
+			length++;
+			node = node.next;
+		}
+
+		int target = length - k - 1;
+		int index = 0;
+		node = list.first;
+
+		// Get target node
+		while (index != target) {
```

Comment:
This requires list to be traversed twice (once to find length and once to find the actual element). How could the code be changed to traverse it only once?

Imagine that someone wants to find 100th element from the end of a list that has 10000000000 elements.